src/github.ts

```diff
@@ -0,0 +1,140 @@
+import * as github from '@actions/github';
+import * as core from '@actions/core';
+import { components } from '@octokit/openapi-types';
+
+const token = core.getInput('token', { required: true });
+const client = github.getOctokit(token);
+
+/**
+ * Get the current context's PR information
+ */
+export function getPr(): components['schemas']['pull-request-minimal'] {
+  const pullRequest = github.context.payload.workflow_run.pull_requests[0];
+  if (!pullRequest) {
+    core.debug(`NO PR FOUND IN CONTEXT (github.content.payload.workflow_run.pull_requests[0]):`);
+    core.setFailed(`PR is undefined. Exiting.`);
+    core.endGroup(); // Gathering inputs...
+    process.exit(1);
+  }
+
+  return pullRequest;
+}
+
+/**
+ * Request reviews on the current context's PR from the specified reviewers.
+ * @param reviewers The list of reviewers to request a review.
+ */
+export async function requestReviewers(reviewers: string[]): Promise<void> {
+  const pr = getPr().number;
+  core.info(`Requesting Reviewers for PR #${pr} to: ${reviewers.join(',')}`);
+
+  const response = await client.rest.pulls.requestReviewers({
+    owner: github.context.repo.owner,
+    repo: github.context.repo.repo,
+    pull_number: pr,
+    reviewers: reviewers,
+  });
+  coreDebugJson(response, `requestReviewers(${pr}, [${reviewers.join(',')}]) > response`);
+}
+
+/**
+ *
+ * Assign the current context's PR to the specified assignees.
+ * @param assignees The list of assignees to assign the PR.
+ */
+export async function setAssignees(assignees: string[]): Promise<void> {
+  const pr = getPr().number;
+  core.info(`Setting Assignees for PR #${pr} to: ${assignees.join(',')}`);
+
+  const response = await client.rest.issues.addAssignees({
+    owner: github.context.repo.owner,
+    repo: github.context.repo.repo,
+    issue_number: pr,
+    assignees: assignees,
+  });
+  coreDebugJson(response, `setAssignees(${pr}, [${assignees.join(',')}]) > response`);
+}
+
+export type CheckRun = components['schemas']['check-run'];
+
+/**
+ * Get the checkruns for the current context's workflow_run for the head SHA.
+ */
+export async function getCheckRuns(): Promise<CheckRun[]> {
+  const ref = github.context.payload.workflow_run.head_sha;
+  const { owner, repo } = github.context.repo;
+
+  core.info(`Retrieving check runs for ${owner}/${repo}@${ref}...`);
+  const checkRuns: CheckRun[] = (
+    await client.rest.checks.listForRef({
+      owner,
+      repo,
+      ref,
+    })
+  ).data.check_runs;
+
+  core.info(`Retrieved ${checkRuns.length} check runs.`);
+  coreDebugJson(checkRuns, 'getCheckRuns() > checkRuns');
+
+  return checkRuns;
+}
+
+/**
+ * Get the required checks for the base ref.
+ */
+export async function getRequiredCheckNames(): Promise<string[] | undefined> {
+  const baseRef = getPr()?.base.ref;
+  core.info(`Base Ref: ${baseRef}`);
+  if (baseRef === undefined) {
```

Comment:
Why check for explicit undefined here, but check using `!` on line 13? You should pick a style and stick with it!

Reply:
No particular reason - is there a convention/style that's more preferential? For the time being I'll switch the first one to check `=== undefined`. (also, similar question to above; any linting tool that'll check for this?)